source/common/network/listen_socket_impl.cc

```diff
@@ -17,6 +18,21 @@ void ListenSocketImpl::doBind() {
   }
 }
 
+TcpListenSocket::TcpListenSocket(const std::string& address, bool bind_to_port) {
```

Comment:
Can TcpListenSocket::TcpListenSocket(uint32_t port, bool bind_to_port) below be implemented in terms of this method? That would reduce the amount of code duplication.